pkg/kubectl/cmd/clusterinfo.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+
+	"github.com/spf13/cobra"
+)
+
+func (f *Factory) NewCmdClusterInfo(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "clusterinfo",
+		Short: "Display cluster info",
+		Long:  "Display IPs of the master and services with label kubernetes.io/cluster-service=true",
+		Run: func(cmd *cobra.Command, args []string) {
+			RunClusterInfo(f, out, cmd)
+		},
+	}
+	return cmd
+}
+
+func RunClusterInfo(f *Factory, out io.Writer, cmd *cobra.Command) {
+	client, err := f.ClientConfig(cmd)
+	checkErr(err)
+	fmt.Fprintf(out, "Kubernetes master is running at %v\n", client.Host)
+
+	mapper, typer := f.Object(cmd)
+	cmdNamespace, err := f.DefaultNamespace(cmd)
+	checkErr(err)
+	args := []string{"pods"}
+	// TODO: use generalized labels once they are implemented (#341)
+	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand(cmd)).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		SelectorParam("kubernetes.io/cluster-service=true").
+		ResourceTypeOrNameArgs(false, args...).
+		Latest()
+
+	b.Do().Visit(func(r *resource.Info) error {
+		pods := r.Object.(*api.PodList).Items
+		for _, pod := range pods {
+			fmt.Fprintf(out, "%v is running at http://%v\n", pod.Labels["name"], pod.Status.HostIP)
```

Comment:
I'm printing label "name" as a temporary solution. The goal is to print value of label "kubernetes.io/cluster-service" introduced in #4620 which will specify the service name (instead of temporary value "true") however it's not possible before generalized labels are implemented #341.

It's implementation of feature request #3290. There is a formal proposal and follow-up discussion as well as in #4417.